libraries/AP_ExternalAHRS/AP_ExternalAHRS_MicroStrain.cpp

```diff
@@ -98,6 +98,10 @@ AP_ExternalAHRS_MicroStrain::AP_ExternalAHRS_MicroStrain(AP_ExternalAHRS *_front
 
 void AP_ExternalAHRS_MicroStrain::update_thread(void)
 {
+    if( uart == nullptr) {
```

Comment:
We shouldn't be starting this thread if uart is nullptr - you've modified the send-text in that clause, and that returns early.

How is uart nullptr here?!

Reply:
This is on my branch, I wasn't able to reproduce this on `master. Closing the PR.

Reply:
Actually, same problem on `master`. 

`AP_ExternalAHRS` has an `update()` method. Here's the call stack. It bypasses the fact that the thread was never created because it calls `update()` regardless of whether the state of the AHRS.

https://github.com/ArduPilot/ardupilot/blob/cc3fe56a088320604aec46fc1a2ca9cd282f5f5a/libraries/AP_ExternalAHRS/AP_ExternalAHRS_MicroStrain.h#L46-L48

https://github.com/ArduPilot/ardupilot/blob/cc3fe56a088320604aec46fc1a2ca9cd282f5f5a/libraries/AP_ExternalAHRS/AP_ExternalAHRS.cpp#L230-L235

https://github.com/ArduPilot/ardupilot/blob/cc3fe56a088320604aec46fc1a2ca9cd282f5f5a/libraries/AP_AHRS/AP_AHRS_External.cpp#L14-L17

https://github.com/ArduPilot/ardupilot/blob/cc3fe56a088320604aec46fc1a2ca9cd282f5f5a/libraries/AP_AHRS/AP_AHRS.cpp#L663-L673

Reply:
The SPI UART thread is lower priority than the main thread due to boost priority. The reason for calling the update() call in the AHRS system is so that the AHRS is guaranteed to get the latest available data. Without this, on plane, it could be 20mS behind.